### PR TITLE
Add compassRuntimeID to OperationStatus GraphQL struct to return information if the runtime is registered in Compass

### PR DESCRIPTION
**Pull request by koala7659**
Implementation of https://github.com/kyma-project/control-plane/issues/3085 **Proposal 1**

Adding new field `compassRuntimeID` to OperationStatus GraphQL struct to return information whether runtime is registered in Compass.

The value returned by this field is now the same as `runtimeID` the Runtime is still registered in Compass.

This is first part of the migration.

Than we will run migration to annotate all Kyma resources with `compass-runtime-id-for-migration` value.

Next step will be [Removing Compass integration code from the Provisioner](https://github.com/kyma-project/control-plane/issues/3088)

Finally the field `compassRuntimeID` will be returned as empty and `runtimeID` will contain some random ID generated locally in Provisioner.

Additionally 

- update base container image to `alpine:3.18.5` to fix busybox vulnerabilities
- update go.mod dependencies of some libraries to latest versions

### CODE REVIEW

**Review by mvshao — APPROVED** [review]
LGTM

File: components/provisioner/Dockerfile [hunk 1]
```diff
@@ -21,7 +21,7 @@ RUN apk add -U --no-cache ca-certificates && update-ca-certificates
 RUN go build -v -o main ./cmd/
 RUN mkdir /app && mv ./main /app/main
 
-FROM eu.gcr.io/kyma-project/external/alpine:3.17.3
+FROM alpine:3.18.5
```
[the remark this answers is not in the document]
Comment (koala7659) [reply components/provisioner/Dockerfile:24]: Warden will only check resulting image

**Review by mvshao — COMMENTED** [review]
LGTM for Provisioner sources

**Comment by piotrmiskiewicz** [comment]
The compassRuntimeID as a field of the RuntimeStatus looks good. This change seems to be backward compatible (just adding new field in the result) so you can merge it as soon as possible.

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**